src/activity_item_handling.cpp

```diff
@@ -2407,15 +2513,22 @@ void generic_multi_activity_handler( player_activity &act, player &p )
                 // its a construction and we need the components.
                 const construction &built_chosen = list_constructions[ *act_info.con_idx ];
                 what_we_need = built_chosen.requirements;
-            } else if( reason == NEEDS_VEH_DECONST ) {
+            } else if( reason == NEEDS_VEH_DECONST || reason == NEEDS_VEH_REPAIR ) {
                 vehicle *veh = veh_pointer_or_null( g->m.veh_at( src_loc ) );
                 // we already checked this in can_do_activity() but check again just incase.
                 if( !veh ) {
                     p.activity_vehicle_part_index = 1;
                     continue;
                 }
                 const vpart_info &vpinfo = veh->part_info( p.activity_vehicle_part_index );
-                const auto &reqs = vpinfo.removal_requirements();
+                requirement_data reqs;
+                if( reason == NEEDS_VEH_DECONST ) {
+                    const requirement_data &local_req = vpinfo.removal_requirements();
```

Comment:
```suggestion
                    reqs = vpinfo.removal_requirements();
```
Same below. No need for a new variable.